azurerm/data_source_client_config.go

```diff
@@ -68,6 +68,9 @@ func dataSourceArmClientConfigRead(d *schema.ResourceData, meta interface{}) err
 	if principal := servicePrincipal; principal != nil {
 		d.Set("service_principal_application_id", principal.AppID)
 		d.Set("service_principal_object_id", principal.ObjectID)
+	} else {
+		d.Set("service_principal_application_id", "")
+		d.Set("service_principal_object_id", "")
 	}
```

Comment:
as discussed IRL I think we should look to replace these in time with more generic `application_id` and `object_id` fields which are available for users too